bser/encode.go

```diff
@@ -224,6 +231,26 @@ func encodeTemplate(buf []byte, r reflect.Value) ([]byte, error) {
 	return appendItem(buf, 0x0b, b), nil
 }
 
+func exportedStructFields(t reflect.Type) ([]reflect.StructField, error) {
+	if t.Kind() != reflect.Struct {
+		return nil, fmt.Errorf("can't get struct fields from value of kind: %s", t.Kind())
+	}
+
+	var (
+		exported  = []reflect.StructField{}
+		numFields = t.NumField()
+	)
+
+	for i := 0; i < numFields; i++ {
+		field := t.Field(i)
+		if len(field.Name) != 0 && field.Name[0] >= 'A' && field.Name[0] <= 'Z' {
+			exported = append(exported, field)
+		}
```

Comment:
I think the more conventional way of checking is by checking if field.PkgPath is empty.  see the comment for StructField.PkgPath here: https://golang.org/pkg/reflect/#StructField

Reply:
@jonasi good call, updated